lib/jekyll/entry_filter.rb

```diff
@@ -86,14 +86,10 @@ def symlink?(entry)
     end
 
     # --
-    # NOTE: Pathutil#in_path? gets the realpath.
-    # @param [<Anything>] entry the entry you want to validate.
-    # Check if a path is outside of our given root.
+    # Check if given path is outside of current site's configured source directory.
     # --
     def symlink_outside_site_source?(entry)
-      !Pathutil.new(entry).in_path?(
-        site.in_source_dir
-      )
+      !File.realpath(entry).start_with?(site.in_source_dir)
```

Comment:
One alternative here might be to check if a constructed path and the real path are not equal?

```ruby
site.in_source_dir(entry) != File.realpath(entry)
```

Reply:
The downside to this is that it results in consumption of additional resources.
`site.in_source_dir(entry)` is more expensive (both in terms of allocations and cpu) than `site.in_source_dir`.

A *better* alternative to *my original proposal here* would be to check directly against `site.source` instead:
```suggestion
      !File.realpath(entry).start_with?(site.source)
```
But since `EntryFilter#symlink_outside_site_source?` method is mostly invoked *only if* `site.safe && File.symlink?(entry)` returns `true`, the difference between `site.source` and `site.in_source_dir` is negligible.

Reply:
Sounds good. :shipit: